### PR TITLE
ci(SPV-992): use dependabot update groups

### PR DESCRIPTION
**Pull request by dorzepowski**

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+      interval: "weekly"
+      day: "monday"
+    labels:
+      - "dependencies"
+    groups:
+      go_modules:
+        update-types:
+          - "minor"
+          - "patch"
+          - "major"
```
Comment (Nazarii-4chain) [inline .github/dependabot.yml:18]: Maybe it would be better to leave major updates as a separate PRs. There is a high chance that major updates will need code changes. WDYT ?

Comment (dorzepowski) [reply .github/dependabot.yml:18]: I would rather start simply with all,
when it will really happen and we will see that it's needed then we can split it.

**Review by wregulski — APPROVED** [review]
